src/main/java/com/example/music/backend/user/controller/UserController.java

```diff
@@ -1,68 +1,40 @@
 import com.example.music.backend.user.service.UserService;
+import lombok.AllArgsConstructor;
 import org.springframework.context.ApplicationEventPublisher;
-import org.springframework.security.crypto.password.PasswordEncoder;
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
-import org.springframework.validation.Errors;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-import org.springframework.web.servlet.ModelAndView;
+
 
 import javax.servlet.http.HttpServletRequest;
 
+import java.net.URI;
+
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 
-@Controller
+@CrossOrigin("http://localhost:3000")
+@AllArgsConstructor
+@RestController
+@RequestMapping("/users")
 public class UserController {
 
     private ApplicationEventPublisher eventPublisher;
     private final UserService service;
-    private final PasswordEncoder encoder;
-
-
-    public UserController(UserService service, ApplicationEventPublisher eventPublisher, PasswordEncoder encoder) {
-        this.service = service;
-        this.eventPublisher = eventPublisher;
-        this.encoder = encoder;
-    }
-
-    @PostMapping(produces = APPLICATION_JSON_VALUE, path = "/user/registration")
-    public ModelAndView create(@ModelAttribute("UserDto") UserDto userDto,
-                               HttpServletRequest request, Errors errors) {
-        ModelAndView mav = new ModelAndView("successRegister", "user", userDto);
-        try {
-            String encodedPassword = encoder.encode(userDto.getPassword());
-            userDto.setPassword(encodedPassword);
-            User saved = service.create(userDto);
-            String appUrl = request.getContextPath();
-            eventPublisher.publishEvent(new OnRegistrationCompleteEvent(saved, request.getLocale(), appUrl));
-        } catch (UserAlreadyExistException uaeEx) {
-            mav.setViewName("registration");
-            mav.addObject("message", "An account for that username/email already exists.");
-        } catch (RuntimeException ex) {
-            mav.setViewName("emailError");
-        }
-        return mav;
-    }
-
-    @GetMapping(path = "/user/registration")
-    public String getPage(Model model) {
-        model.addAttribute("UserDto", new UserDto());
-        return "registration";
-    }
-
-    @GetMapping(path = "/login")
-    public String getLoginPage() {
-        return "login";
-    }
 
-    @GetMapping(path = "/home")
-    public String getHomePage() {
-        return "home";
+    @PostMapping(consumes = APPLICATION_JSON_VALUE, value = "/registration")
+    public ResponseEntity<UserResponse> create(@RequestBody UserDto userDto,
+                                               HttpServletRequest request) {
+        UserResponse response = service.create(userDto);
+        User user = new User();
+        user.setId(response.getId());
+        user.setEmail(userDto.getEmail());
```

Comment:
It looks a bit weird. Why do you use "response" for id and "userDto" for email? Why "response" doesn't suit both cases?